src/PowerShellEditorServices/Services/DebugAdapter/Handlers/LaunchAndAttachHandler.cs

```diff
@@ -345,18 +351,26 @@ public async Task<Unit> Handle(PsesAttachRequestArguments request, CancellationT
                     throw new RpcErrorException(0, "A positive integer must be specified for the RunspaceId field.");
                 }
 
+                _debugStateService.RunspaceId =  runspaceId;
+
                 debugRunspaceCmd = $"\nDebug-Runspace -Id {runspaceId}";
             }
             else
             {
+                _debugStateService.RunspaceId =  1;
+
                 debugRunspaceCmd = "\nDebug-Runspace -Id 1";
             }
 
+            // Clear any existing breakpoints before proceeding
+            await _breakpointService.RemoveAllBreakpointsAsync().ConfigureAwait(continueOnCapturedContext: false);
+
             _debugStateService.WaitingForAttach = true;
             Task nonAwaitedTask = _powerShellContextService
                 .ExecuteScriptStringAsync(debugRunspaceCmd)
                 .ContinueWith(OnExecutionCompletedAsync);
 
+            _jsonRpcServer.SendNotification(EventNames.Initialized);
```

Comment:
The addition of this is what is causing the breakpoint issue for versions before 7.

The debug service actually handles sending this message once a `RunspaceChanged` event hits.  If this is sent before that event fires, then it'll try to invoke the `Get-PSBreakpoint` and etc on the *current* runspace.  Because the current runspace is busy, the first request just hangs waiting on the queue. 

On top of that, the subsequent calls to the breakpoint handler fail because the first call still holds the handle for breakpoints, even though the queue the first call is waiting on is in a different prompt nest frame.

If you comment this line it works

Reply:
That did it!!